src/demo-app/datepicker/datepicker-demo.ts

```diff
@@ -31,11 +36,78 @@ export class DatepickerDemo {
   lastDateInput: Date | null;
   lastDateChange: Date | null;
 
+  monthYearDateControl = new FormControl(moment([2017, 10, 25]));
+  yearDateControl = new FormControl(moment([2017, 10, 25]));
+
   dateFilter =
       (date: Date) => !(date.getFullYear() % 2) && (date.getMonth() % 2) && !(date.getDate() % 2)
 
   onDateInput = (e: MatDatepickerInputEvent<Date>) => this.lastDateInput = e.value;
   onDateChange = (e: MatDatepickerInputEvent<Date>) => this.lastDateChange = e.value;
 
-  dateCtrl = new FormControl();
+  chosenYearHandler(year: number, datepicker: MatDatepicker<Date>) {
```

Comment:
nit: these should probably actually be `MatDatepicker<Moment>` (or `MatDatepicker<any>` since your code doesn't care)

Reply:
I'm gonna use Moment for that just to expose the fact we're using Moment.